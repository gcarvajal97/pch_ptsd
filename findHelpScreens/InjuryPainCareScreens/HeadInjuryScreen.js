import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Linking } from "expo";
import translate from '../../components/translateService';
import colors from '../../colors';

export default class HeadInjuryScreen extends Component {
  render() {
    return (
      <View>
        <Text style={styles.paragraph}>{translate('injuryPainCare.Ch5_p1')}</Text>
        <Text style={styles.paragraph}>{translate('injuryPainCare.Ch5_p2')}</Text>
        <Text style={styles.paragraph}>{translate('injuryPainCare.Ch5_p3')}</Text>

        <Text style={styles.paragraph}>{translate('injuryPainCare.Ch5_p4_part1')}{"\ "}
        <Text 
          testID='hyperlink'
          style={styles.link} 
          onPress={()=> Linking.openURL('https://www.chop.edu/centers-programs/concussion-care-minds-matter')}
          accessible
          accessibilityLabel={translate('injuryPainCare.Ch5_accessibilityLabel')}
          accessibilityHint={translate('injuryPainCare.Ch5_accessibilityHint')}>
          {translate('injuryPainCare.Ch5_click_here')}
        </Text>{translate('injuryPainCare.Ch5_p4_part2')}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  paragraphTitle: {
    fontFamily: 'avenir-heavy',
    fontSize: 20,
    alignSelf: 'stretch',
    textAlign: 'left',
    marginTop: 6,
    marginBottom: 3,
    color: colors.accent
  },
  paragraph: {
    textAlign: 'left',
    fontSize: 14,
    marginVertical: 2,
  },
  paragraphBold: {
    textAlign: 'left',
    fontSize: 14,
    fontFamily: 'avenir-heavy',
    marginVertical: 2,
    marginLeft: 10
  },
  link: {
    fontSize: 14,
    color: colors.accent,
    textDecorationLine: 'underline',
  }
});