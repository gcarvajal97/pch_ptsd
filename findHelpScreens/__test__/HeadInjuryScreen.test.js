import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from 'react';
import HeadInjuryScreen from '../InjuryPainCareScreens/HeadInjuryScreen';
import renderer from 'react-test-renderer';
import {Linking} from 'expo';

configure({ adapter: new Adapter() });
jest.mock('expo', () => ({
    Linking: {
        openURL: jest.fn()
    }
}));


describe("HeadInjuryScreen", () => {
    afterEach(() => {
        jest.clearAllMocks();
      });

    it("renders correctly", () => {
        expect(renderer.create(<HeadInjuryScreen/>).toJSON()).toMatchSnapshot();
    });

    it("clicks external link, opens correct website", () => {
        const wrapper = shallow(<HeadInjuryScreen/>);
        wrapper.findWhere((node) => node.prop('testID') === 'hyperlink').props().onPress();
        expect(Linking.openURL).toHaveBeenCalledWith('https://www.chop.edu/centers-programs/concussion-care-minds-matter');
    })
})