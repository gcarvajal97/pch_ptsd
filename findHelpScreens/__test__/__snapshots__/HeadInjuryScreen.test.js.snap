// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadInjuryScreen renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
    Head injury is a broad term that describes injuries to the scalp, skull, brain, and tissue and blood vessels in the child’s head. Some head injuries are also called brain injury, or traumatic brain injury (TBI), depending on the extent of the head trauma. Concussion is the most common type of TBI.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
    A concussion is caused by a blow or jolt to the head or body that causes the brain to shake. The shaking can cause the brain not to work normally and can result in serious side effects. If your child hits her head, be aware of the concussion symptoms to look for, which are commonly physical (such as a headache, nausea, or dizziness), sleep-related, difficulty concentrating/remembering, or involve mood disruption.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
    Initial treatment for a concussion is rest, both mental and physical, which allows the brain time to heal. Every child’s injury and recovery is unique, but with guidance from a doctor, your child can slowly return to school and to play following a concussion.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
    For comprehensive information about concussion, including how a concussion is diagnosed, prevention strategies, returning to school and play, and support resources, visit
     
    <Text
      accessibilityHint="Navigates to a website about concussions"
      accessibilityLabel="Click here for concussion tips"
      accessible={true}
      onPress={[Function]}
      style={
        Object {
          "color": "#0077C8",
          "fontSize": 14,
          "textDecorationLine": "underline",
        }
      }
      testID="hyperlink"
    >
      Concussion Care for Kids: Minds Matter
    </Text>
    , a website developed by an interdisciplinary team of concussion experts at The Children's Hospital of Philadelphia.
  </Text>
</View>
`;
